include/mxnet/c_lapack_api.h

```diff
@@ -0,0 +1,80 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file c_lapack_api.h
+ * \brief Unified interface for LAPACK calls from within mxnet. 
+ *  Purpose is to hide the platform specific differences.
+ */
+#ifndef MXNET_C_LAPACK_API_H_
+#define MXNET_C_LAPACK_API_H_
+
+// Manually maintained list of LAPACK interfaces that can be used
+// within MXNET. Conventions:
+//    - Interfaces must be compliant with lapacke.h in terms of signature and
+//      naming conventions so wrapping a function "foo" which has the
+//      signature
+//         lapack_int LAPACKE_foo(int, char, lapack_int, float* , lapack_int)
+//      within lapacke.h should result in a wrapper with the following signature
+//         int MXNET_LAPACK_foo(int, char, int, float* , int)
+//    - It is ok to assume that matrices are stored in contiguous memory
+//      (which removes the need to do special handling for lda/ldb parameters
+//      and enables us to save additional matrix transpositions around
+//      the fortran calls).
+//    - It is desired to add some basic checking in the C++-wrappers in order
+//      to catch simple mistakes when calling these wrappers.
+//    - Must support compilation without lapack-package but issue runtime error in this case.
+
```

Comment:
May we add one line of comment stipulating the expected layout of input/output matrix (always row-major vs column-major)?